src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpSensorBridge.h

```diff
@@ -351,16 +350,152 @@ class YarpSensorBridge : public ISensorBridge,
+
+    /**
+     * Get all motors' PWMs
+     * @param[out] motorPWMs all motors' PWM in ampere
+     * @param[out] receiveTimeInSeconds time at which the measurement was received
+     * @warning the size is decided at the configuration and remains fixed,
+     * and internal checks must be done at the implementation level by the Derived class.
+     * This means that the user must pass a resized argument "motorPWMs" to this method
+     * @return true/false in case of success/failure
+     */
+    bool getMotorPWMs(Eigen::Ref<Eigen::VectorXd> motorPWMs,
+                           OptionalDoubleRef receiveTimeInSeconds = {}) final;
+
+    /**
+     * Get motor torque in Nm
+     * @param[in] jointName name of the joint
+     * @param[out] jointTorque motor torque in Nm
+     * @param[out] receiveTimeInSeconds time at which the measurement was received
+     * @return true/false in case of success/failure
+     */
+    bool getJointTorque(const std::string& jointName,
+                          double& jointTorque,
+                          OptionalDoubleRef receiveTimeInSeconds = {}) final;
+
+    /**
+     * Get all motors' torques in Nm
+     * @param[out] jointTorques all motors' torque in Nm
+     * @param[out] receiveTimeInSeconds time at which the measurement was receivedW
+     * @warning the size is decided at the configuration and remains fixed,
+     * and internal checks must be done at the implementation level by the Derived class.
+     * This means that the user must pass a resized argument "jointTorques" to this method
+     * @return true/false in case of success/failure
+     */
+    bool getJointTorques(Eigen::Ref<Eigen::VectorXd> jointTorques,
+                           OptionalDoubleRef receiveTimeInSeconds = {}) final;
+
+    /**
+     * Get pid position in rad
+     * @param[in] jointName name of the joint
+     * @param[out] pidPosition pid position in radians
+     * @param[out] receiveTimeInSeconds time at which the measurement was received
+     * @return true/false in case of success/failure
+     */
+    virtual bool getPidPosition(const std::string& jointName,
+                                  double& pidPosition,
+                                  OptionalDoubleRef receiveTimeInSeconds = {}) final;
+
+    /**
+     * Get all pid positions in rad
+     * @param[out] pidPositions all pid positions in radians
+     * @param[out] receiveTimeInSeconds time at which the measurement was received
+     * @warning the size is decided at the configuration and remains fixed,
+     * and internal checks must be done at the implementation level by the Derived class.
+     * This means that the user must pass a resized argument "pidPositions" to this methodW
+     * @return true/false in case of success/failure
+     */
+    virtual bool getPidPositions(Eigen::Ref<Eigen::VectorXd> pidPositions,
+                                  OptionalDoubleRef receiveTimeInSeconds = {}) final;
```

Comment:
What is the position of a PID?

Reply:
Opening a new issue for this

Reply:
https://github.com/dic-iit/bipedal-locomotion-framework/issues/367